Lib/test/test_netrc.py

```diff
@@ -1,45 +1,126 @@
+            os.chmod(netrc_file, mode=mode)
+        return netrc_file
+
+
+class NetrcBuilder:
+    """Utility class to construct and load `netrc.netrc` instances using
+    different configuration scenarios.
+    """
+
+    @staticmethod
+    def use_default_netrc_in_home(*args, **kwargs):
+        """Load an instance of netrc using the default `.netrc` file from the
+        user's home directory.
+        """
+        with NetrcEnvironment() as helper:
+            helper.environ.unset("NETRC")
+            helper.environ.set("HOME", helper.tmpdir)
+            helper.generate_netrc(*args, **kwargs)
+            return netrc.netrc()
+
+    @staticmethod
+    def use_netrc_envvar(*args, **kwargs):
+        """Load an instance of the netrc using the `.netrc` file specified by
+        the `NETRC` environment variable.
+        """
+        with NetrcEnvironment() as helper:
+            netrc_file = helper.generate_netrc(*args, **kwargs)
+            helper.environ.set("NETRC", netrc_file)
+            return netrc.netrc()
+
+    @staticmethod
+    def use_file_argument(*args, **kwargs):
+        """Load an instance of `.netrc` file using the file as argument.
+        """
+        with NetrcEnvironment() as helper:
+            # Just to stress a bit more the test scenario, the NETRC envvar
+            # will contain rubish information which shouldn't be used
+            helper.environ.set("NETRC", "not-a-file.netrc")
+            netrc_file = helper.generate_netrc(*args, **kwargs)
+            return netrc.netrc(netrc_file)
+
+    @staticmethod
+    def get_all_scenarios():
+        """Return all `.netrc` loading scenarios as callables.
+
+        This method is useful for iterating through all d ways the
+        `.netrc` file can be located.
+        """
+        return (NetrcBuilder.use_default_netrc_in_home,
+                NetrcBuilder.use_netrc_envvar,
+                NetrcBuilder.use_file_argument)
+
 
 class NetrcTestCase(unittest.TestCase):
+    ALL_NETRC_FILE_SCENARIOS = NetrcBuilder.get_all_scenarios()
 
-    def make_nrc(self, test_data):
-        test_data = textwrap.dedent(test_data)
-        mode = 'w'
-        if sys.platform != 'cygwin':
-            mode += 't'
-        with open(temp_filename, mode, encoding="utf-8") as fp:
-            fp.write(test_data)
-        try:
-            nrc = netrc.netrc(temp_filename)
-        finally:
-            os.unlink(temp_filename)
-        return nrc
-
-    def test_toplevel_non_ordered_tokens(self):
-        nrc = self.make_nrc("""\
+    @subTests('make_nrc', ALL_NETRC_FILE_SCENARIOS)
+    def test_toplevel_non_ordered_tokens(self, make_nrc):
+        nrc = make_nrc("""\
             machine host.domain.com password pass1 login log1 account acct1
             default login log2 password pass2 account acct2
             """)
-        self.assertEqual(nrc.hosts['host.domain.com'], ('log1', 'acct1', 'pass1'))
+        self.assertEqual(nrc.hosts['host.domain.com'],
```

Comment:
Sorry I should have said: only wrap lines under 80 chars if it's new code, not if it's old code.